File: benchmarks/go/matrix_sum.go
package main

import "fmt"

const N = 500

func main() {
	matrix := make([][]int64, N)
	for i := range matrix {
		matrix[i] = make([]int64, N)
	}

	for i := 0; i < N; i++ {
		for j := 0; j < N; j++ {
			matrix[i][j] = int64(i*N + j)
		}
	}

	var sum int64 = 0
	for i := 0; i < N; i++ {
		for j := 0; j < N; j++ {
			sum += matrix[i][j]
		}
	}

	fmt.Println(sum)
}

File: benchmarks/rust/matrix_sum.rs
const N: usize = 500;

fn main() {
    let mut matrix = vec![vec![0i64; N]; N];

    for i in 0..N {
        for j in 0..N {
            matrix[i][j] = (i * N + j) as i64;
        }
    }

    let mut sum: i64 = 0;
    for i in 0..N {
        for j in 0..N {
            sum += matrix[i][j];
        }
    }

    println!("{}", sum);
}
